homeassistant/components/vallox/__init__.py

```diff
@@ -0,0 +1,259 @@
+            'method': 'async_set_profile_fan_speed_away',
+            'schema': SERVICE_SCHEMA_SET_PROFILE_FAN_SPEED},
+        SERVICE_SET_PROFILE_FAN_SPEED_BOOST: {
+            'method': 'async_set_profile_fan_speed_boost',
+            'schema': SERVICE_SCHEMA_SET_PROFILE_FAN_SPEED},
+}
+
+DEFAULT_FAN_SPEED_HOME = 50
+DEFAULT_FAN_SPEED_AWAY = 25
+DEFAULT_FAN_SPEED_BOOST = 65
+
+
+async def async_setup(hass, config):
+    """Set up the client and boot the platforms."""
+    conf = config[DOMAIN]
+    host = conf.get(CONF_HOST)
+    name = conf.get(CONF_NAME)
+
+    client = Vallox(host)
+    state_proxy = ValloxStateProxy(hass, client)
+    service_handler = ValloxServiceHandler(client, state_proxy)
+
+    hass.data[DOMAIN] = {
+        'client': client,
+        'state_proxy': state_proxy,
+        'name': name
+    }
+
+    for vallox_service in SERVICE_TO_METHOD:
+        schema = SERVICE_TO_METHOD[vallox_service]['schema']
+        hass.services.async_register(DOMAIN, vallox_service,
+                                     service_handler.async_handle,
+                                     schema=schema)
+
+    # Fetch initial state once before bringing up the platforms.
+    await state_proxy.async_update(None)
+
+    hass.async_create_task(
+        async_load_platform(hass, 'sensor', DOMAIN, {}, config))
+    hass.async_create_task(
+        async_load_platform(hass, 'fan', DOMAIN, {}, config))
+
+    async_track_time_interval(hass, state_proxy.async_update, SCAN_INTERVAL)
+
+    return True
+
+
+class ValloxStateProxy:
+    """Helper class to reduce websocket API calls."""
+
+    def __init__(self, hass, client):
+        """Initialize the proxy."""
+        self._hass = hass
+        self._client = client
+        self._metric_cache = {}
+        self._profile = None
+        self._valid = False
+
+    def fetch_metric(self, metric_key):
+        """Return cached state value."""
+        _LOGGER.debug("Fetching metric key: %s", metric_key)
+
+        if not self._valid:
+            raise OSError("Device state out of sync.")
+
+        if metric_key not in vlxDevConstants.__dict__:
+            raise KeyError("Unknown metric key: {}".format(metric_key))
+
+        return self._metric_cache[metric_key]
+
+    def get_profile(self):
+        """Return cached profile value."""
+        _LOGGER.debug("Returning profile")
+
+        if not self._valid:
+            raise OSError("Device state out of sync.")
+
+        return PROFILE_TO_STR_REPORTABLE[self._profile]
+
+    async def async_update(self, event_time):
+        """Fetch state update."""
+        _LOGGER.debug("Updating Vallox state cache")
+
+        try:
+            self._metric_cache = await self._hass.async_add_executor_job(
+                self._client.fetch_metrics)
+            self._profile = await self._hass.async_add_executor_job(
+                self._client.get_profile)
+            self._valid = True
+
+        except IOError as io_err:
```

Comment:
We can except OSError instead. IOError is an alias of OSError.

https://docs.python.org/3/library/exceptions.html#IOError

> The following exceptions are kept for compatibility with previous versions; starting from Python 3.3, they are aliases of OSError.

```py
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.3.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: try: 
   ...:     raise IOError 
   ...: except OSError: 
   ...:     print("Caught the IOError") 
   ...:                                                                                                                                                                                                       
Caught the IOError

In [2]:
```